src/permission/fs_permission.cc

```diff
@@ -85,7 +86,7 @@ void FSPermission::Apply(const std::string& allow, PermissionScope scope) {
       }
       return;
     }
-    GrantAccess(scope, res);
+    GrantAccess(scope, std::string(res.data(), res.size()));
```

Comment:
This is creating a new string for every parameter while the current behaviour is passing by reference. Why should we have this change?

Reply:
For this particular usage:

`res` was a `std::string` and now it is `std::string_view`. Previously, we were creating a new string for every parameter, while splitting the string, right now we are creating new string for every parameter except `*`.

Reply:
Actually, you are creating a `std:::string_view` (res) for every parameter, and a `std::string` for all parameters except `*`.

Reply:
`std::string_view` provides a readonly access to an existing `std::string` without making a copy. So, creation of a std::string_view does not have any performance cost.